src/main/java/net/neoforged/neoforge/attachment/AttachmentType.java

```diff
@@ -58,165 +53,131 @@
      */
-    public static <T> Builder<T> builder(Function<IAttachmentHolder, T> defaultValueConstructor) {
+    public static <T, P extends IAttachmentHolder> Builder<T, P> builder(Function<P, T> defaultValueConstructor) {
         return new Builder<>(defaultValueConstructor);
     }
 
     /**
-     * Create a builder for an attachment type that uses {@link INBTSerializable} for serialization.
-     * Other kinds of serialization can be implemented using {@link #builder(Supplier)} and {@link Builder#serialize(IAttachmentSerializer)}.
-     *
-     * <p>See {@link #serializable(Function)} for attachments that want to capture a reference to their holder.
-     */
-    public static <S extends Tag, T extends INBTSerializable<S>> Builder<T> serializable(Supplier<T> defaultValueSupplier) {
-        return serializable(holder -> defaultValueSupplier.get());
-    }
-
-    /**
-     * Create a builder for an attachment type that uses {@link INBTSerializable} for serialization.
-     * Other kinds of serialization can be implemented using {@link #builder(Supplier)} and {@link Builder#serialize(IAttachmentSerializer)}.
+     * Creates a builder for an attachment type.
      *
      * <p>This overload allows capturing a reference to the {@link IAttachmentHolder} for the attachment.
-     * To obtain a specific subtype, the holder can be cast.
      * If the holder is of the wrong type, the constructor should throw an exception.
-     * See {@link #serializable(Supplier)} for an overload that does not capture the holder.
+     * See {@link #builder(Supplier)} for an overload that does not capture the holder.
+     *
+     * @param parentClass             The attachment handler's parent/containing type class, i.e. Entity, LevelChunk, BlockEntity
+     * @param defaultValueConstructor A constructor for a new default value of this attachment type.
+     * @param <P>                     The attachment handler's parent/containing type, i.e. Entity, LevelChunk, BlockEntity
+     * @param <T>                     Data type of the attachment
      */
-    public static <S extends Tag, T extends INBTSerializable<S>> Builder<T> serializable(Function<IAttachmentHolder, T> defaultValueConstructor) {
-        return builder(defaultValueConstructor).serialize(new IAttachmentSerializer<S, T>() {
-            @Override
-            public T read(IAttachmentHolder holder, S tag, HolderLookup.Provider provider) {
-                var ret = defaultValueConstructor.apply(holder);
-                ret.deserializeNBT(provider, tag);
-                return ret;
-            }
-
-            @Nullable
-            @Override
-            public S write(T attachment, HolderLookup.Provider provider) {
-                return attachment.serializeNBT(provider);
-            }
-        });
+    public static <T, P extends IAttachmentHolder> Builder<T, P> builder(Class<P> parentClass, Function<P, T> defaultValueConstructor) {
```

Comment:
This overload with the `parentClass` arg is entirely unnecessary. For example, where someone might use
```java
AttachmentType.builder(ChunkAccess.class, chunk -> new ChunkMutableInt(chunk, 0))
```
With this overload, they can instead just use
```java
AttachmentType.builder((ChunkAccess chunk) -> new ChunkMutableInt(chunk, 0))
```
With the normal overload and all the type inference will just work -- I just tested this to be sure. So I'd toss this one out entirely